openapi-core/src/main/java/io/ballerina/openapi/core/generators/schema/TypeGeneratorUtils.java

```diff
@@ -262,24 +265,41 @@ private static RecordFieldWithDefaultValueNode getRecordFieldWithDefaultValueNod
      * @param fieldSchema Schema for data type
      * @return {@link MetadataNode}
      */
-    public static AnnotationNode generateConstraintNode(Schema<?> fieldSchema) {
-
-        if (fieldSchema instanceof StringSchema) {
-            StringSchema stringSchema = (StringSchema) fieldSchema;
-            // Attributes : maxLength, minLength
-            return generateStringConstraint(stringSchema);
-        } else if (fieldSchema instanceof IntegerSchema || fieldSchema instanceof NumberSchema) {
-            // Attribute : minimum, maximum, exclusiveMinimum, exclusiveMaximum
-            return generateNumberConstraint(fieldSchema);
-        } else if (fieldSchema instanceof ArraySchema) {
-            ArraySchema arraySchema = (ArraySchema) fieldSchema;
-            // Attributes: maxItems, minItems
-            return generateArrayConstraint(arraySchema);
+    public static AnnotationNode generateConstraintNode(String typeName, Schema<?> fieldSchema) {
+        if (isSupportConstraint(typeName, fieldSchema)) {
+            if (fieldSchema instanceof StringSchema) {
+                StringSchema stringSchema = (StringSchema) fieldSchema;
+                // Attributes : maxLength, minLength
+                return generateStringConstraint(stringSchema);
+            } else if (fieldSchema instanceof IntegerSchema || fieldSchema instanceof NumberSchema) {
+                // Attribute : minimum, maximum, exclusiveMinimum, exclusiveMaximum
+                return generateNumberConstraint(fieldSchema);
+            } else if (fieldSchema instanceof ArraySchema) {
+                ArraySchema arraySchema = (ArraySchema) fieldSchema;
+                // Attributes: maxItems, minItems
+                return generateArrayConstraint(arraySchema);
+            }
+            // Ignore Object, Map and Composed schemas.
+            return null;
         }
-        // Ignore Object, Map and Composed schemas.
         return null;
     }
 
+    public static boolean isSupportConstraint(String typeName, Schema schema) {
+
+        boolean isConstraintSupport = schema.getNullable() != null && schema.getNullable() ||
```

Comment:
Shall we change the `isConstraintSupport` variable to something like this ex: `constraintNotFollowed`, `avoidsConstraint` to align with the condition checks?

Reply:
Normally in Java it is a best practice to start a boolean variable with `is`. At the same time I was reusing the name defined [here](https://github.com/ballerina-platform/openapi-tools/blob/364cd5d8f34321af31c7b5283e7793530552d35c/openapi-core/src/main/java/io/ballerina/openapi/core/generators/schema/TypeGeneratorUtils.java#L175) for the same purpose.

Reply:
Changed to `isConstraintAllowed`

Reply:
Changed to `isConstraintNotAllowed` in https://github.com/ballerina-platform/openapi-tools/pull/1301/commits/5b1a80e8798f333b69f746f2c5bb6162f9792b73